### ISSUE TITLE
[Rule change request] `vue/no-unused-properties`: New option to take `expose` into account

### ISSUE THREAD

**Issue by kawadah** [vuejs/eslint-plugin-vue#2385]
**What rule do you want to change?**

[vue/no-unused-properties](https://eslint.vuejs.org/rules/no-unused-properties.html)

**Does this change cause the rule to produce more or fewer warnings?**

Fewer false positives.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option (maybe `checkExpose`).

Currently, `vue/no-unused-properties` will report an error if the property is not used anywhere else in the same SFC.

With the new option set to `true`, ESLint should ignore properties defined in `expose` (added in Vue 3.2), and report errors on properties that are unused in the same SFC **and** not defined in `expose`.

:warning: One thing to consider: When `expose` is `undefined` (which means expose everything), how should the new option behave?
(I don't have a reasonable idea, so I'll leave it to everyone)

**Please provide some example code that this change will affect:**

```js
// .eslintrc.js
module.exports = {
  rules: {
    "vue/no-unused-properties": ["warn", {
      groups: ["methods"]
    }]
  }
}
```

```vue
<template>
  <button @click="foo()">Button</button>
</template>

<script>
export default {
  expose: [
    'bar'
  ],
  methods: {
    foo() {},
    bar() {},
    baz() {}
  }
}
</script>
```

**What does the rule currently do for this code?**

Report 2 warnings.

```log
'bar' of method found, but never used.
'baz' of method found, but never used.
``` 

**What will the rule do after it's changed?**

Report 1 warning (since `bar` is defined in `expose`).

```
'baz' of method found, but never used.
```

**Additional context**

> This rule cannot check for use of properties by other components (e.g. mixins, property access via $refs)

The reason my team at work decided not to enable this rule is because of the false positives, even if we ARE using the properties through `$refs`.

If we take advantage of `expose`, we can reduce the false positives and make the rule more useful.

**Comment by ota-meshi** [comment]
I can't reproduce the problem. Currently works as you expect.
https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJx1VMuO2zAM/BXDl7aLCAF6DNJF0d+o9yDLtKNdWXIpKY8N8u8dyYkdp+nFCYcUpRk+zqVyDZWbchuoH4wM9FrZotjWMQRni5/KaPXxoypb575+q8rXXxnfrkc/Yrfru4MwvWI9BPxfvxTkjbahqMp9pLV1ItroqREDu4E4aPJVuSl+V+VBsq3KVXFOV3fs4uAz3lPYuQZRb5W9vBUv68rScXAcioZaGU0YTyTMUzoBoyi+1JK/pL9vq/S9JtmMsUWRmRTny2o0Ebw0P7OZLHwuieCVUrkqORry5eacCSnX92SDaDSTCnoPFb+vsufdHwWYerGX7DOqnPWBowqOhY9gn9EW1vW0sxnpKARiwRQijwhkk/5kVVKt154EHUlF5Ll5lQTYkDKSZcrjJ4eR3uOw192UCm8eJJOw1IlPYjfjtvknFE9+hklwTvF6evXoCewMXt7R8QY2VMeuI55tQ4GSKhMCLYTkzi+A8eU99TXx0kMGZHW7wD7odB+EhkUzZlluKBo0nBaGUDvIpZLW+bKlc5CpChM3Oj7IQMfAUtTOGZI23RRunlYaE3bo4G43QRHFW57vjKuleQB1nzr7EbQWT3xWXW330uhmFHyYUIYdESwOOx3ID1JNKhgHUV2LRiKl/V3t0FYg0mjb/U8WSwfhT33tzIQ461QACbxN9/glr+Qw3eXqd+hizJQhB98MtHJw4YTa1VGboGdWTCNXmgHQE5nHFOTJtA86+Wdz43eycQeAGD0NUo2wsr9Lg0nwqftYzg0UdtqLmjCZdDeoeXdh58wGNKecssca0lhzNPuYJHSszR3kZUui1RaSnB5QjKK8n6Qr7IYEQlik0ha1mf15iRpZ0yzvFZz2TYIgE6HitVQfTC0x2bkXbk6Mito9gstSHnQYI5j+RCwrcdJkmozENIzeyvHtYz+msrp2WoXPtj6cl8tf7Dk0lQ==
